csharp/src/Ice/UtilInternal/Collections.cs

```diff
@@ -7,83 +7,21 @@ namespace Ice.UtilInternal;
 
 public sealed class Collections
 {
-    public static bool SequenceEquals(ICollection seq1, ICollection seq2)
+    // Just like Enumerable.SequenceEqual except it also handles null sequences.
+    // TODO: the generated hash code is not consistent with the "null is equivalent to empty" logic implemented by
```

Comment:
In the previous (3.7) implementation, a null collection field does not compare equal to an empty collection field, even though they both encoded as empty.